luwak/src/test/java/uk/co/flax/luwak/TestSlowLog.java

```diff
@@ -54,11 +53,41 @@ public Weight createWeight(IndexSearcher searcher, boolean needsScores) {
                         } catch (InterruptedException e) {
                             throw new RuntimeException(e);
                         }
-                        return new RandomAccessWeight(this) {
+                        return new ConstantScoreWeight(this, boost) {
+                            @Override
+                            public final Scorer scorer(LeafReaderContext context) throws IOException {
+                              final Bits matchingDocs = getMatchingDocs(context);
+                              if (matchingDocs == null || matchingDocs instanceof Bits.MatchNoBits) {
+                                return null;
+                              }
+                              final DocIdSetIterator approximation = DocIdSetIterator.all(context.reader().maxDoc());
+                              final TwoPhaseIterator twoPhase = new TwoPhaseIterator(approximation) {
+
+                                @Override
+                                public boolean matches() throws IOException {
+                                  final int doc = approximation.docID();
+
+                                  return matchingDocs.get(doc);
+                                }
+
+                                @Override
+                                public float matchCost() {
+                                  return 10; // TODO: use some cost of matchingDocs
+                                }
+                              };
+
+                              return new ConstantScoreScorer(this, score(), twoPhase);
+                            }
+
                             protected Bits getMatchingDocs(LeafReaderContext context) throws IOException {
                                 return new Bits.MatchAllBits(context.reader().maxDoc());
                             }
 
+                            @Override
+                            public boolean isCacheable(LeafReaderContext ctx) {
```

Comment:
This method would be removed in Lucene 7 IIRC but is needed as long as Luwak is built with Lucene 6.